### ISSUE TITLE
New rule: Require explicit static on all parts

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/StyleCopAnalyzers#2986]
```cs
public static partial class C
{
        public static class C1
        {
            [Test]
            public void M1()
            {
            }
        }
}
```

```cs
public partial class [|C|]
{
        public static class C2
        {
            [Test]
            public void M2()
            {
            }
        }
}
```

The second part is static but adding explicit static modifier would make it clearer.

**Comment by vweijsters** [comment]
It does make sense to me to have such a rule. It will definitely improve readability when working with partial classes.

**Comment by sharwell** [comment]
I'm torn since this one makes it difficult to change the class from static to non-static. For cases like ___, there is a compiler error we can watch for and automatically fix all cases. That doesn't apply here though.

I suppose we could implement this rule we two code fixes: one code fix would _add_ `static` to a part where the diagnostic is reported, and the other part would _remove_ `static` from all other parts.
